### DISCUSSION TITLE
[11.0.0] :nth-last-child(1) does not seem to be working properly on ItemsRepeater

### DISCUSSION THREAD

**Discussion by Mrcubix** [AvaloniaUI/Avalonia#12147] [Q&A]
For some reasons, the selector `:nth-last-child(1)` seems to only be evaluated properly at first, any additions done after the Window is shown will result in an incorrect behavior.

in this exemple, the last element is supposed to have a margin of 0, so that the scrollbar doesn't show up until another element is added.

<p align="center">
    <img src="https://github.com/AvaloniaUI/Avalonia/assets/39861216/ef04a4ae-fb24-4338-bb8d-9203ec03fcb1">
</p>

the same behavior can be obtained using this repo: https://github.com/Mrcubix/Avalonia-Last-Nth-Child-Selector-Issue

<p align="center">
    <img src="https://github.com/AvaloniaUI/Avalonia/assets/39861216/caa5a00a-77d4-467d-a4c2-6fcd3a199a8b" alt="Avalonia bug where the ItemsRepeater control only evaluates the nth-last-child(1) selector at initial render" />
</p>

about the same styling works for `ItemsControl`, however, `ItemsControl` suffer from a worse issue explained here #12137

**Comment by StefanKoell** [comment]
I'm seeing the same issue. Having a style selector targeting all classes, for example works fine even for elements inside an items repeater but as soon as you use `:nth-last-child(1)` is doesn't work anymore. I'm using 11.0.5-rc1 and the issue is still present.

**Reply by timunie** [reply]
this is not an issue tracker but a discussion. So if one finds it's an actual bug in latest nightly, and there is no open issue for it please open one.

**Reply by Mrcubix** [reply]
Now it is #13270
